src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -1041,4 +1041,41 @@ private void repaintIfBackgroundsDiffer(JComponent comp) {
             }
         }
     }
+
+    // Toggle flag for drawing the mnemonic state
+    private static boolean isMnemonicHidden = true;
+
+    /**
+     * Sets the state of the hide mnemonic flag. This flag is used by the
+     * component UI delegates to determine if the mnemonic should be rendered.
+     * This method is a non operation if the underlying operating system
+     * does not support the mnemonic hiding feature.
+     *
+     * @param hide true if mnemonics should be hidden
+     * @since 23
+     */
+    public static void setMnemonicHidden(boolean hide) {
+        if (UIManager.getBoolean("Button.showMnemonics")) {
+            // Do not hide mnemonics if the UI defaults do not support this
+            isMnemonicHidden = false;
+        } else {
+            isMnemonicHidden = hide;
+        }
```

Comment:
Can we use or even re-use the logic in Windows L&F to display and hide menu mnemonics?

Is this fix limited to menu mnemonics? Does it apply to all mnemonics in components?

Reply:
> Can we use or even re-use the logic in Windows L&F to display and hide menu mnemonics?

For windows behavior is different than linux.
https://github.com/openjdk/jdk/pull/18992#issuecomment-2100033545

I don't think that the logic in Windows L&F can be reused here.

> Is this fix limited to menu mnemonics? Does it apply to all mnemonics in components?

As of now verified with the menu elements.